ext/gmp/gmp.c

```diff
@@ -47,6 +47,12 @@
 #define GMP_BIG_ENDIAN    (1 << 3)
 #define GMP_NATIVE_ENDIAN (1 << 4)
 
+#if SIZEOF_SIZE_T == 4
+#define GMP_ALLOC_MAXBITS (ULONG_MAX / GMP_NUMB_BITS)
+#else
+#define GMP_ALLOC_MAXBITS INT_MAX
+#endif
```

Comment:
The logic to avoid overflow in `_mpz_realloc()` (libgmp 6.3.0) is:
````c
  if (sizeof (mp_size_t) == sizeof (int))
    {
      if (UNLIKELY (new_alloc > ULONG_MAX / GMP_NUMB_BITS))
	MPZ_OVERFLOW;
    }
  else
    {
      if (UNLIKELY (new_alloc > INT_MAX))
	MPZ_OVERFLOW;
    }
````
Assuming that `mp_size_t` is actually `size_t`, the simplification seems to be correct (we're assuming `sizeof(int) == 4` anyway). However, I do not understand how `ULONG_MAX` fits into the mix. That only works under the assumption that `ULONG_MAX == UINT_MAX` for 32bit platforms (which is what php-src assumes anyway), but why don't they use `UINT_MAX` in the first place? I think we should.